client/src/fluid/Fluid.res

```diff
@@ -1175,130 +1174,132 @@ let caretTargetForStartOfExpr = (astPartId: id, ast: FluidAST.t): CT.t =>
     )
   }
 
-/* caretTargetForStartOfPattern returns a caretTarget representing caret
- placement at the very start of the expression in `pattern` */
-let caretTargetForStartOfPattern = (pattern: fluidPattern): CT.t =>
-  switch pattern {
-  | PVariable(id, _) => {astRef: ARPattern(id, PPVariable), offset: 0}
-  | PConstructor(id, _, _) => {astRef: ARPattern(id, PPConstructor), offset: 0}
-  | PInteger(id, _) => {astRef: ARPattern(id, PPInteger), offset: 0}
-  | PBool(id, _) => {astRef: ARPattern(id, PPBool), offset: 0}
-  | PString(id, _) => CT.forPPStringOpenQuote(id, 0)
-  | PCharacter(_, _) =>
+@ocaml.doc("returns a caretTarget representing caret placement at the very
+  start of the expression in `matchPattern`")
+let caretTargetForStartOfMP = (matchPattern: fluidMatchPattern): CT.t =>
+  switch matchPattern {
+  | MPVariable(id, _) => {astRef: ARMPattern(id, MPPVariable), offset: 0}
+  | MPConstructor(id, _, _) => {astRef: ARMPattern(id, MPPConstructor), offset: 0}
+  | MPInteger(id, _) => {astRef: ARMPattern(id, MPPInteger), offset: 0}
+  | MPBool(id, _) => {astRef: ARMPattern(id, MPPBool), offset: 0}
+  | MPString(id, _) => CT.forMPPStringOpenQuote(id, 0)
+  | MPCharacter(_, _) =>
     recover(
-      "echar unsupported in caretTargetForStartOfPattern",
+      "echar unsupported in caretTargetForStartOfMP",
       {FluidCursorTypes.CaretTarget.astRef: ARInvalid, offset: 0},
     )
-  | PFloat(id, _, _, _) => {astRef: ARPattern(id, PPFloat(FPWhole)), offset: 0}
-  | PNull(id) => {astRef: ARPattern(id, PPNull), offset: 0}
-  | PBlank(id) => {astRef: ARPattern(id, PPBlank), offset: 0}
-  | PTuple(id, _, _, _) => {astRef: ARPattern(id, PPTuple(TPOpen)), offset: 0}
+  | MPFloat(id, _, _, _) => {astRef: ARMPattern(id, MPPFloat(FPWhole)), offset: 0}
+  | MPNull(id) => {astRef: ARMPattern(id, MPPNull), offset: 0}
+  | MPBlank(id) => {astRef: ARMPattern(id, MPPBlank), offset: 0}
+  | MPTuple(id, _, _, _) => {astRef: ARMPattern(id, MPPTuple(TPOpen)), offset: 0}
   }
 
-/* caretTargetForEndOfPattern returns a caretTarget representing caret
- * placement at the very end of the expression in `pattern`.
- *
- * The concept of "very end" is related to an understanding of the
- * tokenization of the ast, even though this function doesn't explicitly depend
- * on any tokenization functions. */
-let rec caretTargetForEndOfPattern = (pattern: fluidPattern): CT.t =>
-  switch pattern {
-  | PVariable(id, varName) => {
-      astRef: ARPattern(id, PPVariable),
+@ocaml.doc(
+  "returns a caretTarget representing caret placement at the very end
+  of the expression in `matchPattern`.
+
+  The concept of "
```

Comment:
Weird formatting here

Reply:
Ah - that's what happens when you use `"`s in an `@ocaml.doc()`, it seems.